hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/index/IndexingCatchupTaskFactory.java

```diff
@@ -36,11 +38,14 @@ public static IndexingCatchupTask createCatchupTask(List<HoodieIndexPartitionInf
                                                       HoodieTableMetadataWriter metadataWriter,
                                                       List<HoodieInstant> instantsToIndex,
                                                       Set<String> metadataCompletedInstants,
-                                                      HoodieTableMetaClient metaClient,
+                                                      HoodieTable table,
                                                       HoodieTableMetaClient metadataMetaClient,
                                                       String currentCaughtupInstant,
                                                       TransactionManager transactionManager,
                                                       HoodieEngineContext engineContext) {
+    HoodieTableMetaClient metaClient = table.getMetaClient();
+    HoodieHeartbeatClient heartbeatClient = new HoodieHeartbeatClient(table.getStorage(), metaClient.getBasePath().toString(), table.getConfig().getHoodieClientHeartbeatIntervalInMs(),
```

Comment:
would this just check for heartbeats..or actually do any heartbeats.. i.e write heartbeats.. ? just making sure

Reply:
unless we do heartbeatClient.start(instantTime), no heart beats will be emitted.

Reply:
should we create the heartbeat in the constructor of RunIndexActionExecutor. 
which is where transaction manager is created and passed along